inference-engine/src/cldnn_engine/ops/parameter.cpp

```diff
@@ -37,7 +37,7 @@ void CreateParameterOp(Program& p, const std::shared_ptr<ngraph::op::v0::Paramet
 
     cldnn::tensor dataTensor;
     cldnn::tensor::value_type batch = (p.m_max_batch <= 1)
-                                    ? (inputDims.size() > 3 ? TensorValue(inputDims[0]) : 1)
+                                    ? ((inputDims.size() > 3 || inputDims.size() == 2) ? TensorValue(inputDims[0]) : 1)
                                     : TensorValue(p.m_curBatch);
```

Comment:
Looks like this may be simplified to:
`    cldnn::tensor::value_type batch = (p.m_max_batch <= 1) ? TensorValue(inputDims[0]) : TensorValue(p.m_curBatch);`

And I suppose it makes sense to fix same potential issue for `inputDims.size() == 3` case

Reply:
The condition was simplified. 
As for `inputDims.size() == 3` case, we support only CHW 3d layout, so dynamic batch value shouldn't be applied to it.

Reply:
Okay, I got it